utils/systemd/manager.go

```diff
@@ -0,0 +1,169 @@
+	privateExecutor
+	dirs   systemdDirs
+	logger logging.Logger
+}
+
+// SystemdManagerOption is a type used to configure the [SystemdManager]
+// returned from [NewSystemdManager].
+type SystemdManagerOption func(*SystemdManager)
+
+// WithExecutor configures the created [SystemdManager] with a custom
+// [SystemdExecutor] implementation. Should only be used for testing.
+func WithExecutor(executor SystemdExecutor) SystemdManagerOption {
+	return func(manager *SystemdManager) {
+		manager.privateExecutor = executor
+	}
+}
+
+// WithDirs configures the created [SystemdManager] with custom service file
+// search paths. Should only be used for testing.
+func WithDirs(serviceFileDir, fallbackFileDir string) SystemdManagerOption {
+	return func(manager *SystemdManager) {
+		manager.dirs.serviceFileDir = serviceFileDir
+		manager.dirs.fallbackFileDir = fallbackFileDir
+	}
+}
+
+func NewSystemdManager(logger logging.Logger, opts ...SystemdManagerOption) *SystemdManager {
+	manager := &SystemdManager{
+		logger:          logger,
+		privateExecutor: realSystemdExecutor{},
+		dirs: systemdDirs{
+			serviceFileDir:  defaultServiceFileDir,
+			fallbackFileDir: defaultFallbackFileDir,
+		},
+	}
+	for _, opt := range opts {
+		opt(manager)
+	}
+	return manager
+}
+
+// InstallService creates or updates a service file. If a service file with the
+// same name exists in an deprecated location it will also automatically remove
+// the old file. It returns the path the installed service file if it exists.
+// It also returns true if the install was successful and the service did not
+// already exist.
+func (s *SystemdManager) InstallService(serviceName string, serviceFileContents []byte) (string, bool, error) {
+	serviceFileName := serviceName + ".service"
+	serviceFilePath, removeOldFile, err := s.getServiceFilePath(serviceFileName)
+	if err != nil {
+		return "", false, err
+	}
+
+	// Track if the service file already existed an any way. Use this later to
+	// avoid reenabling a service the user disabled.
+	newInstall := !removeOldFile
+	if newInstall {
+		_, err = os.Stat(serviceFilePath)
+		newInstall = err != nil
+	}
+
+	s.logger.Infof("writing systemd service file to %s", serviceFilePath)
+
+	newFile, err := utils.WriteFileIfNew(serviceFilePath, serviceFileContents)
+	if err != nil {
+		return "", false, errors.Wrapf(err, "writing systemd service file %s", serviceFilePath)
+	}
+
+	if removeOldFile {
+		oldPath := filepath.Join(s.dirs.fallbackFileDir, serviceFileName)
+		s.logger.Warn("Removing system service file %s in favor of vendor file at %s", oldPath, serviceFilePath)
+		s.logger.Warn("If you customized this file, please run 'systemctl edit viam-agent' and create overrides there")
+		if err := os.RemoveAll(oldPath); err != nil {
+			s.logger.Warn(errors.Wrapf(err, "removing old service file %s, please delete manually", oldPath))
+		}
+	}
+
+	if newFile {
+		if err := s.DaemonReload(); err != nil {
+			return "", false, err
+		}
```

Comment:
Are you missing `s.Enable()` here? Trying to follow along [from where](https://github.com/viamrobotics/agent/blob/8a6c789ff1633262cfb68baf3249f6e2f1e1fb69/agent.go#L121-L129) you moved this code.